podcast/2.17.md

```diff
@@ -0,0 +1,487 @@
+
+And that was the extent of it, that was what I took away from those conversations. But I remember putting this idea into my head about, "Oh, incremental parsing that makes sense that you want to do, but that isn't a Salt." And then I could see that once I switched to using VIM, I could understand, like, oh, VIM doesn't really do syntax analysis. There's this really rough syntax highlighting and then the rest is all from this power of image based on text processing, not really code processing. So it made sense to me then, once I switched back from using IDEs to using VIM.
+
+So I put this idea in my head like, huh, there's just this unsolved problem that we all don't have a solution to, which is the idea of having a fast editor that also has the same syntax awareness that the JetBrains ones have. That was really interesting to me. It was my first realization that like, "Oh yeah, these tools that we all use as programmers? They're changing all the time and you can just make your own stuff." A couple years into working at Pivotal, I was looking for something hard to work on or something different than the app development cycle that I was in.
+
+And I still had that thought about, "Oh, incremental parsing. Like, is that a thing?" And so I started on the side. Mostly in the mornings, when I was taking the bus from Oakland to San Francisco, I would do my research about this idea. What would it take to make that happen? I had this vision from early on. I really admired these software projects where it seemed like they were solving a dense, hard problem and making it sort of fade away. Like, once it had been solved by this library, you just didn't even have to worry about it anymore. SQLite was the number one thing like that to me. It was this C library that if you wanted a managed, structured data you just use SQLite and it completely solved a problem. It was then a non-problem. And it was this really gnarly thing that it solved and other things like that.
+
+Like libgit2 was a similar thing where it's like, if you want to do good stuff just use libgit2 and you can do it from any language that has bindings to the C library.
+
+I had it in my mind that, okay, well, I should figure out how parsing works. I didn’t know that because I never went to computer science school. Then once I do know that, then I have to find out how you do incremental parsing. There seemed to be papers about it. I didn’t understand them yet, but I'll read them once I understand parsing. And then if you could just do it as a C library, then it would be solved and you could put it in an editor like VIM. That was the end game at the time. If it's a C Library, you could even use it in VIM or any editor. You could portably solve it.
```

Comment:
```suggestion
I had it in my mind that, okay, well, I should figure out how parsing works. I didn’t know that because I never went to computer science school. Then once I do know that, then I have to find out how you do incremental parsing. There seemed to be papers about it. I didn’t understand them yet, but I'll read them once I understand parsing. And then if you could just do it as a C library, then it would be solved and you could put it in an editor like Vim. That was the end game at the time. If it's a C Library, you could even use it in Vim or any editor. You could portably solve it.
```